src/bsoncxx/config/prelude.hpp

```diff
@@ -15,4 +15,12 @@
 #include <bsoncxx/config/config.hpp>
 #include <bsoncxx/export.hpp>
 
-#define BSONCXX_INLINE inline __attribute__ ((__visibility__("hidden"), __always_inline__))
+#pragma push_macro("MONGOCXX_INLINE")
```

Comment:
I think you need to do these before you include the headers above. Otherwise, you are just pushing our definitions. In the legacy C++ driver, we also undef after we push so that our decl doesn't conflict. Checkout undef_macros and redef_macros in the legacy driver.